frontend/src/app/testQueue/TestCardForm/TestCardForm.utils.tsx

```diff
@@ -0,0 +1,221 @@
+            disease.supportedDisease.name !== MULTIPLEX_DISEASES.COVID_19
+        ).length > 0
+    );
+  }
+  return false;
+};
+
+export const isDeviceFluOnly = (deviceId: string, devicesMap: DevicesMap) => {
+  if (devicesMap.has(deviceId)) {
+    const supportedDiseaseTests =
+      devicesMap.get(deviceId)?.supportedDiseaseTestPerformed ?? [];
+
+    return (
+      supportedDiseaseTests.length > 0 &&
+      supportedDiseaseTests.every(
+        (disease) =>
+          disease.supportedDisease.name === MULTIPLEX_DISEASES.FLU_A ||
+          disease.supportedDisease.name === MULTIPLEX_DISEASES.FLU_B
+      )
+    );
+  }
+  return false;
+};
+
+export const hasAnySupportedDiseaseTests = (
+  deviceId: string,
+  devicesMap: DevicesMap
+) => {
+  if (devicesMap.has(deviceId)) {
+    const supportedDiseaseTests =
+      devicesMap.get(deviceId)?.supportedDiseaseTestPerformed ?? [];
+    return supportedDiseaseTests.length > 0;
+  }
+  return false;
+};
+
+// when other diseases are added, update this to use the correct AOE for that disease
+export const useAOEFormOption = (deviceId: string, devicesMap: DevicesMap) => {
+  // some devices don't have any supported disease tests saved because historically they only supported COVID
+  // this is often seen in some of the dev environments
+  if (!hasAnySupportedDiseaseTests(deviceId, devicesMap)) {
+    return AOEFormOption.COVID;
+  }
+  return isDeviceFluOnly(deviceId, devicesMap)
+    ? AOEFormOption.NONE
+    : AOEFormOption.COVID;
+};
+
+export const convertFromMultiplexResponse = (
+  responseResult: QueriedTestOrder["results"]
+): MultiplexResultInput[] => {
+  return responseResult.map((result) => ({
+    diseaseName: result.disease?.name,
+    testResult: result.testResult,
+  }));
+};
+
+export const areAOEAnswersComplete = (
+  formState: TestFormState,
+  whichAOE: AOEFormOption
+) => {
+  if (whichAOE === AOEFormOption.COVID) {
+    const isPregnancyAnswered = !!formState.covidAOEResponses.pregnancy;
+    const hasNoSymptoms = formState.covidAOEResponses.noSymptoms;
+    if (formState.covidAOEResponses.noSymptoms === false) {
+      const symptoms = parseSymptoms(formState.covidAOEResponses.symptoms);
+      const areSymptomsFilledIn = Object.values(symptoms).some((x) =>
+        x.valueOf()
+      );
+      const isSymptomOnsetDateAnswered =
+        !!formState.covidAOEResponses.symptomOnset;
+      return (
+        isPregnancyAnswered &&
+        !hasNoSymptoms &&
+        areSymptomsFilledIn &&
+        isSymptomOnsetDateAnswered
+      );
+    }
+    return isPregnancyAnswered && hasNoSymptoms;
+  }
+  return true;
+};
+
+export const showTestResultDeliveryStatusAlert = (
+  deliverySuccess: boolean | null | undefined,
+  patient: SomeoneWithName
+) => {
+  if (deliverySuccess === false) {
+    const { title, body } = {
+      ...ALERT_CONTENT[
+        QUEUE_NOTIFICATION_TYPES.DELIVERED_RESULT_TO_PATIENT__FAILURE
+      ](patient),
+    };
+    return showError(body, title);
```

Comment:
is there a reason we are surfacing the errors as an alert at the top of the page as opposed to the inline form error pattern we use elsewhere? 🤔 
<img width="1210" alt="Screenshot 2023-10-20 at 15 52 07" src="https://github.com/CDCgov/prime-simplereport/assets/20211771/1d5a0040-9667-49b5-82fd-d51e8aa2ed90">

Reply:
I'll defer to design on the specific reason but I believe this was related to handling when we have multiple radio groups for a multiplex device. Before it gets enabled in prod, we'll be updating the test card design based on upcoming user feedback so we could address it in that future story